core/test/org/javarosa/xpath/expr/test/XPathPathExprTest.java

```diff
@@ -105,6 +115,24 @@ public void doTests() {
         testEval("/data/places/country[@id ='one']/name", instance, null, "Singapore");
     }
 
+    /**
+     * Some simple xpath expressions with multiple predicates that operate over
+     * nodesets.
+     */
+    private void testNestedMultiplicities() {
+        FormParseInit fpi = new FormParseInit("/test_nested_multiplicities.xml");
+        FormDef fd = fpi.getFormDef();
+        FormEntryModel fem = fpi.getFormEntryModel();
+
+        testEval("/data/bikes/manufacturer/model[@id='pista']/@color",
+                fd.getInstance(), null, "seafoam");
+        testEval("join(' ', /data/bikes/manufacturer[@american='yes']/model[.=1]/@id)",
+                fd.getInstance(), null, "karate-monkey vamoots");
+        // fails because of [model=1]:
+        testEval("join(' ', /data/bikes/manufacturer[@american='yes'][model=1]/model/@id)",
+                fd.getInstance(), null, new XPathTypeMismatchException());
```

Comment:
Are we 100% sure this exception type is appropriate? General XPathException might fit better?